### PR TITLE
Start enabling better use of webpack and module system

### PR DESCRIPTION
**Pull request by tomratcliffe**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /data*
 *.DS_Store
 node_modules/*
+build
```
Comment (MindaugasDukauskas) [inline .gitignore]: We should remove this line as we will need builded version

File: js/Genoverse.js [hunk 2]
```diff
@@ -109,10 +114,14 @@ var Genoverse = Base.extend({
...
+        browser.loadedPlugins[plugin] = browser.loadedPlugins[plugin] || 'script';
+        deferred.resolve(arg);
+      }
+
       function getCSS() {
-        function done() {
-          browser.loadedPlugins[plugin] = browser.loadedPlugins[plugin] || 'script';
-          deferred.resolve(arg);
+        if(runningInModule || true) {
+          return done();
```
Comment (MindaugasDukauskas) [inline js/Genoverse.js]: You need to check 
On line 137
`$.getScript(js, getCSS);`
As it tries to pull scripts when in storybook

File: js/Genoverse.js [hunk 3]
```diff
@@ -109,10 +114,14 @@ var Genoverse = Base.extend({
...
+      function done() {
+        browser.loadedPlugins[plugin] = browser.loadedPlugins[plugin] || 'script';
+        deferred.resolve(arg);
+      }
+
       function getCSS() {
-        function done() {
-          browser.loadedPlugins[plugin] = browser.loadedPlugins[plugin] || 'script';
-          deferred.resolve(arg);
+        if(runningInModule || true) {
```
Comment (MindaugasDukauskas) [inline js/Genoverse.js]: Shouldnt it be only `if(runningInModule) {`

File: index.js [hunk 4]
```diff
@@ -81,3 +86,11 @@ require(__dirname + '/js/plugins/karyotype.js');
 require(__dirname + '/js/plugins/resizer.js');
 require(__dirname + '/js/plugins/tooltips.js');
 require(__dirname + '/js/plugins/trackControls.js');
+
+// CSS:
+require(__dirname + '/css/genoverse.css');
+
+// Plugins:
+require('./plugins.index.js')
```
Comment (MindaugasDukauskas) [inline index.js:85]: So if you load this file. Lines above it with plugins is not needed anymore?

File: webpack.config.js [hunk 5]
```diff
@@ -1,22 +1,56 @@
...
   mode    : 'production',
-  entry   : __dirname + '/index.js',
+  entry   : {
+    genoverse: __dirname + '/index.js'
+  },
   target  : [ 'web', 'es5' ],
-  output  : { filename: 'genoverse.min.js', path: __dirname + '/js' },
+  output  : {
+    filename: '[name].min.js',
+    path: __dirname + '/build',
```
Comment (MindaugasDukauskas) [inline webpack.config.js:14]: I had to add these two lines:
```
library: 'Genoverse',
libraryTarget: 'commonjs2'
```
As for some reason package doesn't export genoverse without it

**Review by MindaugasDukauskas — CHANGES_REQUESTED** [review]
Left a few comments. Apart from that looks good

**Comment by tomratcliffe** [comment]
Thanks @MindaugasDukauskas, I addressed the comments, could you have another look please?

File: js/Genoverse.js [hunk 6]
```diff
@@ -133,7 +134,7 @@ var Genoverse = Base.extend({
 
       if (browser.loadedPlugins[plugin] || $('script[src="' + js + '"]').length) {
         getCSS();
-      } else {
+      } else if (!runningInModule) {
```
Comment (MindaugasDukauskas) [inline js/Genoverse.js:137]: You still need to add `done()`

Comment (MindaugasDukauskas) [reply js/Genoverse.js:137]: Could be it works like that. But I think done resolves `deffered`. Could be wrong

**Review by MindaugasDukauskas — APPROVED** [review]
Added one comment. Not sure if its valid. But looks good

File: js/Genoverse.js (hunk 6 above)
Comment (tomratcliffe) [reply js/Genoverse.js:137]: I checked - it still works correctly in storybook, it doesn't try to re-load the plugins

Comment (MindaugasDukauskas) [reply js/Genoverse.js:137]: 👍

**Pull request merged**